sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubConsumer.java

```diff
@@ -30,15 +36,22 @@
  * @see EventHubClient#createConsumer(String, String, EventPosition, EventHubConsumerOptions)
  */
 public class EventHubConsumer implements Closeable {
+    private static final AtomicReferenceFieldUpdater<EventHubConsumer, SynchronousEventSubscriber> SUBSCRIBER =
+        AtomicReferenceFieldUpdater.newUpdater(EventHubConsumer.class, SynchronousEventSubscriber.class,
+            "eventSubscriber");
+
+    private final ClientLogger logger = new ClientLogger(EventHubConsumer.class);
+    private final AtomicLong idGenerator = new AtomicLong();
+
     private final EventHubAsyncConsumer consumer;
-    private final EventHubConsumerOptions options;
+    private final Duration timeout;
+    private volatile SynchronousEventSubscriber eventSubscriber;
 
     EventHubConsumer(EventHubAsyncConsumer consumer, EventHubConsumerOptions options) {
-        this.consumer = Objects.requireNonNull(consumer, "'consumer' cannot be null.");
-        this.options = Objects.requireNonNull(options, "'options' cannot be null.");
+        Objects.requireNonNull(options, "'options' cannot be null.");
 
-        //TODO (conniey): Keep track of the last sequence number as each method invoked.
-        this.consumer.receive().windowTimeout(options.prefetchCount(), this.options.retry().tryTimeout());
+        this.consumer = Objects.requireNonNull(consumer, "'consumer' cannot be null.");
+        this.timeout = options.retry().tryTimeout();
```

Comment:
Is it weird to have an argument of type `EventHubConsumerOptions` that you take only the retry try timeout out of and then discard the rest? Could that be the argument into the constructor instead (or doesn't it really matter as this is non-public API)?